tests/ops/fixtures/saas/shopify_fixtures.py

```diff
@@ -100,3 +104,153 @@ def shopify_dataset_config(
     )
     yield dataset
     dataset.delete(db=db)
+
+
+@pytest.fixture(scope="function")
+def shopify_erasure_data(
+    shopify_connection_config, shopify_erasure_identity_email, shopify_secrets
+) -> Generator:
+    """
+    Creates a dynamic test data record for erasure tests.
+    Yields customer ID as this may be useful to have in test scenarios
+    """
+    base_url = f"https://{shopify_secrets['domain']}"
+    faker = Faker()
+    firstName = faker.first_name()
+    lastName = faker.last_name()
+    body = {
+        "customer": {
+            "first_name": firstName,
+            "last_name": lastName,
+            "email": shopify_erasure_identity_email,
+            "verified_email": True,
+            "addresses": [
+                {
+                    "address1": "123 Test",
+                    # "city": faker.city(),
+                    "city": "Toronto",
+                    "province": "ON",
+                    "zip": "66777",
+                    # "zip": faker.postcode(),
+                    "last_name": lastName,
+                    "first_name": firstName,
+                    "country": "Canada",
+                    # "country": faker.country(),
+                }
+            ],
+        }
+    }
+    headers = {"X-Shopify-Access-Token": f"{shopify_secrets['access_token']}"}
+    customers_response = requests.post(
+        url=f"{base_url}/admin/api/2022-07/customers.json", json=body, headers=headers
+    )
+    customer = customers_response.json()
+    assert customers_response.ok
+    error_message = f"customer with email {shopify_erasure_identity_email} could not be added to Shopify"
+    poll_for_existence(
+        customer_exists,
+        (shopify_erasure_identity_email, shopify_secrets),
+        error_message=error_message,
+    )
+
+    # Create Order
+    body = {
+        "order": {
+            "email": shopify_erasure_identity_email,
+            "fulfillment_status": "fulfilled",
+            "send_receipt": True,
+            "financial_status": "paid",
+            "send_fulfillment_receipt": True,
+            "line_items": [
+                {
+                    "product_id": 6923717967965,
+                    "name": "Short leeve t-shirt",
+                    "title": "Short sleeve t-shirt",
+                    "price": 10,
+                    "quantity": 1,
+                }
+            ],
+        }
+    }
+    orders_response = requests.post(
+        url=f"{base_url}/admin/api/2022-07/orders.json", json=body, headers=headers
+    )
+
+    assert orders_response.ok
+    order = orders_response.json()
+    order_id = order["order"]["id"]
+
+    # Get Blog
+    blogs_response = requests.get(
+        url=f"{base_url}/admin/api/2022-07/blogs.json", headers=headers
+    )
+    assert blogs_response.ok
+    blog = blogs_response.json()["blogs"][1]
```

Comment:
any particular reason we get a blog rather than creating one for the test fixture? i haven't really looked at the API and i don't have much context, so it may be a dumb suggestion/infeasible. but ideally we'd not have to rely on any data being present for the tests to pass, so if we could create the blog in the fixture (and then clean it up afterward), i think it'd be preferred. let me know what you think.

Reply:
Basically in Shopify we can't filter specific blogs by our customer email so creating one blog in fixture and using it to create article was actually resulting in creating articles in all existing blogs. I had a session with Adrian to verify this when I identified this issue so we agreed to create one blog for testing there already and then we should just get that blog to create new article for erasure endpoint. I can explain this offline in case this isn't clear.

Reply:
makes sense - thanks for explaining!